08-projects.Rmd

```diff
@@ -5195,25 +5231,25 @@ Relevant topics: [group by](#sql-examples), [count](#sql-examples)
 ```
 
-##### 6. In (4) we wrote a query that finds the average `mean_temperature_f` by `zip_code`. What if we want to tack on to our results information from each row in the `station` table based on the `zip_code`s? To do, use an INNER JOIN. INNER JOIN combines tables based on specified fields, and returns only rows where there is a match in both the "left" and "right" tables.
+##### 6. In (4) we wrote a query that finds the average `mean_temperature_f` by `zip_code`. What if we want to tack on our results in (4) to information from each row in the `station` table based on the `zip_code`s? To do, use an INNER JOIN. INNER JOIN combines tables based on specified fields, and returns only rows where there is a match in both the "left" and "right" tables.
 
 **Hint:** Use the query from (4) as a sub query within your solution.
 
-**Relevant topics:** [inner join](#sql-inner-join), subqueries, [aliasing](#sql-aliasing)
+**Relevant topics:** [INNER JOIN](#sql-inner-join), subqueries, [aliasing](#sql-aliasing)
 
 ```{block, type="bbox"}
 **Item(s) to submit:**
 
 - SQL query used to solve this problem.
 ```
 
-##### 7. In (5) we eluded that the `zip_code`s in the `trip` table aren't very consistent. Users can enter a zip code when using the app. This means that `zip_code` can be from anywhere in the world! With that being said, if the `zip_code` is one of the 5 `zip_code`s for which we have weather data (from question 4), we can add that weather information to matching rows of the `trip` table. In (6) we used an INNER JOIN to append some weather information to each row in the `station` table. For this question, write a query that performs an INNER JOIN and appends weather data from the `weather` table to the trip data from the `trip` table. Limit your solution to 5 lines. 
+##### 7. In (5) we eluded that the `zip_code`s in the `trip` table aren't very consistent. Users can enter a zip code when using the app. This means that `zip_code` can be from anywhere in the world! With that being said, if the `zip_code` is one of the 5 `zip_code`s for which we have weather data (from question 4), we can add that weather information to matching rows of the `trip` table. In (6) we used an INNER JOIN to append some weather information to each row in the `station` table. For this question, write a query that performs an INNER JOIN and appends weather data from the `weather` table to the trip data from the `trip` table. Limit your output to 5 lines. 
 
-**Hint:** You will want to wrap your dates and datetimes in [sqlite's `date` function](https://www.sqlitetutorial.net/sqlite-date-functions/sqlite-date-function/) prior to comparison.
+**Important note:** Notice that the weather data has about 1 row of weather information for each date and each zip code. This means you may have to join your data based on multiple constraints instead of just 1 like in (6). In the `trip` table, you can use `start_date` for for the date information.
```

Comment:
I switch the hint and this note as I feel that students will better understand the hint after realizing they need to consider dates. 

I also include `start_date` in the note to nudge the students to use that column (instead of `end_date`). There are five records in `trip` where the start and end dates are not the same day, as illustrated by the following query.
```
SELECT (DATE(end_date) - DATE(start_date)) AS "Date Difference", COUNT(*)
FROM trip
GROUP BY (DATE(end_date) - DATE(start_date));

-- Date Difference|COUNT(*)
-- 0|669954
-- 1|5
```

Reply:
I agree with this change as well, thanks!